crates/model/src/order.rs

```diff
@@ -196,7 +187,7 @@ impl OrderBuilder {
 
 /// An order as provided to the orderbook by the frontend.
 #[serde_as]
-#[derive(Eq, PartialEq, Clone, Deserialize, Debug, Serialize, Hash)]
+#[derive(Eq, PartialEq, Copy, Clone, Deserialize, Debug, Serialize, Hash)]
```

Comment:
Should this really be `Copy`? The struct is pretty big.

Reply:
Copy is about whether the type can be copied by just copying bits without extra logic. This is the case for order so it makes sense to be copy. At one point in this PR I noticed I had to call clone a bunch so I made this change.
This is also why at your other comment I turned that into a reference, because there is nothing to be gained by passing it by value. For Copy types moving them is the same as copying them.

Reply:
Alright. I was under the impression that a type should only be `Copy` if it's also small-ish.

Reply:
[TIL](https://doc.rust-lang.org/std/marker/trait.Copy.html#when-should-my-type-be-copy):

> Generally speaking, if your type _can_ implement `Copy`, it should.

Reply:
Keep in mind is that for Copy types moving them is the same as copying them. So if you have a big type that is Copy (regardless of whether you actually derive Copy) and you take it by value as a function argument, then it will always be copied completely. So whether your type is Copy and whether it should be passed by value is not related which I think is where the confusion stems from. It just so happens that many types you would pass by value because they are small are also Copy.

Unfortunately (?) Rust uses the same trait Copy to signal several things:
1. Type can be copied by just copying bits
2. Type does not have a custom Drop implementation
3. Type automatically gets copied in several instances like when dereferencing.

To me the most important part is 1 and then 2 and 3 is a slight downside for big Copy types because you might accidentally copy them when not intended. Arguably there should have been several traits for this.